lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -155,7 +155,7 @@ def given_error
 
           backtrace = format_backtrace(@actual_error.backtrace)
           [
-            ", got #{@actual_error.inspect} with backtrace:",
+            ", got #{RSpec::Support::ObjectInspector.inspect(@actual_error)} with backtrace:",
```

Comment:
Seeing the extra verbosity here makes me think we should change this matcher (and others where you had to do this) to subclass `RSpec::Matchers::BaseMatcher`.  The matchers that do not subclass that base class aren't intended to specifically avoid subclassing it; it's just that they didn't have a reason to subclass it before, but I think they do now.